File: python/Single Number II.py
class Solution(object):
    def singleNumber(self, nums):
        answer = 0
        single = [0 for x in range(32)]
        for num in nums:
            for i in xrange(32):
                if (num & (1 << i)):
                    single[i] = (single[i] + 1) % 3
        
        for i in range(0, 32):
            if i == 31 and single[i]:
                answer = -((1 << 31) - answer)
            else:
                answer |= single[i] * (1 << i)
        return answer

File: cpp/Single Number II.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        vector<int> bits(32, 0);
        for (int i = 0; i < nums.size(); i++) {
            for (int j = 0; j < 32; j++) {
                bits[j] += (nums[i] >> j) & 1;
            }
        }
        int answer = 0;
        for (int j = 0; j < 32; j++) {
            answer += (bits[j] % 3) << j;
        }
        return answer;
    }
};
